crates/storage/provider/src/change.rs

```diff
@@ -0,0 +1,1429 @@
+//! Wrapper around revms state.
+use rayon::slice::ParallelSliceMut;
+use reth_db::{
+    cursor::{DbCursorRO, DbCursorRW, DbDupCursorRO, DbDupCursorRW},
+    models::{AccountBeforeTx, BlockNumberAddress},
+    tables,
+    transaction::{DbTx, DbTxMut},
+};
+use reth_interfaces::db::DatabaseError;
+use reth_primitives::{
+    bloom::logs_bloom, keccak256, proofs::calculate_receipt_root_ref, Account, Address,
+    BlockNumber, Bloom, Bytecode, Log, Receipt, StorageEntry, H256, U256,
+};
+use reth_revm_primitives::{
+    db::states::{
+        BundleState, PlainStateReverts, PlainStorageChangeset, PlainStorageRevert, RevertToSlot,
+        StateChangeset,
+    },
+    into_reth_acc, into_revm_acc,
+    primitives::AccountInfo,
+};
+use reth_trie::{
+    hashed_cursor::{HashedPostState, HashedPostStateCursorFactory, HashedStorage},
+    StateRoot, StateRootError,
+};
+use std::{collections::HashMap, iter::Peekable};
+
+pub use reth_revm_primitives::db::states::OriginalValuesKnown;
+
+/// Bundle state of post execution changes and reverts
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
+pub struct BundleStateWithReceipts {
+    /// Bundle state with reverts.
+    bundle: BundleState,
+    /// Receipts.
+    receipts: Vec<Vec<Option<Receipt>>>,
+    /// First block of bundle state.
+    first_block: BlockNumber,
+}
+
+/// Type used to initialize revms bundle state.
+pub type BundleStateInit =
+    HashMap<Address, (Option<Account>, Option<Account>, HashMap<H256, (U256, U256)>)>;
+
+/// Types used inside RevertsInit to initialize revms reverts.
+pub type AccountRevertInit = (Option<Option<Account>>, Vec<StorageEntry>);
+
+/// Type used to initialize revms reverts.
+pub type RevertsInit = HashMap<BlockNumber, HashMap<Address, AccountRevertInit>>;
```

Comment:
I still really want to get rid of these,

In any case they should not be public and it should be explained what these represent, but I'd prefer actual types over aliases

Reply:
Would be work related to this: https://github.com/paradigmxyz/reth/issues/4614